net/nimble/host/src/ble_sm.c

```diff
@@ -892,6 +895,72 @@ ble_sm_key_dist(struct ble_sm_proc *proc,
     }
 }
 
+static int
+ble_sm_chk_store_overflow_once(int is_our_sec, uint16_t conn_handle)
```

Comment:
two times in ble_sm_chk_store_overflow():
1) if check for our_keys fails 
2) if check for their_keys fails

Third place is in "ble_store_write" 

Just to make sure I was clear in my first comment :) Idea is to keep ble_sm_chk_store_overflow_once(obj_type)  - maybe change name to ble_sm_chk_store_overflow_by_type() and this function just check if there is required space.
And second helper  ble_store_overflow_event(obj_type, event_code, void *event_data) which just sends event.